src/Processors/Transforms/MergeJoinTransform.h

```diff
@@ -193,11 +193,8 @@ class AllJoinState : boost::noncopyable
 class FullMergeJoinCursor : boost::noncopyable
 {
 public:
-    explicit FullMergeJoinCursor(const Block & sample_block_, const SortDescription & description_)
-        : sample_block(sample_block_.cloneEmpty())
-        , desc(description_)
-    {
-    }
+    /// Contract for FullMergeJoinCursor is that it should be created with materialized block
+    explicit FullMergeJoinCursor(const Block & sample_block_, const SortDescription & description_);
```

Comment:
currently this is the only place when it is created: https://github.com/ClickHouse/ClickHouse/blob/80f39c44d3ac1b6150d6c0952f3e4209fc971c4c/src/Processors/Transforms/MergeJoinTransform.cpp#L43